lib/model/model.go

```diff
@@ -1095,9 +1109,9 @@ func (m *Model) ResumeDevice(device protocol.DeviceID) {
 
 func (m *Model) IsPaused(device protocol.DeviceID) bool {
 	m.pmut.Lock()
-	paused := m.devicePaused[device]
-	m.pmut.Unlock()
-	return paused
+	defer m.pmut.Unlock()
+
+	return m.devicePaused[device]
```

Comment:
```
root@audrius:~# cat test.go
package main

func foo() int {
        return 0
}

func main() {
        foo()
}
root@audrius:~# go build -gcflags '-m' test.go
# command-line-arguments
./test.go:3: can inline foo
./test.go:7: can inline main
./test.go:8: inlining call to foo
root@audrius:~# cat test.go
package main

func foo() int {
        defer func(){}()
        return 0
}

func main() {
        foo()
}
root@audrius:~# go build -gcflags '-m' test.go
# command-line-arguments
./test.go:4: can inline foo.func1
./test.go:4: foo func literal does not escape
```

Yet it seems that anything that involves a mutex escapes to heap killing inlining.